model/src/main/kotlin/utils/ProvenanceFileStorage.kt

```diff
@@ -23,21 +23,21 @@ import java.io.File
 import org.ossreviewtoolkit.model.KnownProvenance
 
 /**
- * A storage which associates a file with a [KnownProvenance]
+ * A generic storage interface that associates a [KnownProvenance] with a belonging file.
  */
 interface ProvenanceFileStorage {
     /**
-     * Return whether a file corresponding to [provenance] exists.
+     * Return whether a file belonging to [provenance] exists.
      */
     fun hasFile(provenance: KnownProvenance): Boolean
 
     /**
-     * Associate the [file] to [provenance]. Overwrites any existing file corresponding to [provenance].
+     * Associate [provenance] with the given [file]. Overwrites any existing association for [provenance].
      */
     fun putFile(provenance: KnownProvenance, file: File)
 
     /**
-     * Return the file corresponding to [provenance] or null if no such file exists. The returned file is a
+     * Return the file belonging to [provenance], or null if there is no such file. The returned file is a
```

Comment:
Should we use "associated with" instead of "belonging" to use same terminology as in `putFile` ?

(Same for docs of `hasFile()`)

Reply:
Having Kotlin's `associateWith` in mind, the thing that comes after "with" should be the value, not the key. That is, I'd prefer *not* to say "Return the file associated with ..." as that would again swap the key / value meaning. To be aligned with Kotlin wording, I'll consider rewording to "associated by".